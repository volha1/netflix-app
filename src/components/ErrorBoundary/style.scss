@import '../../common/scss/base.scss';

.error-message-block {
  min-height: 60vh;
  padding: 50px 0px;

  h1 {
    font-weight: 400;
    text-align: center;
  }
}
