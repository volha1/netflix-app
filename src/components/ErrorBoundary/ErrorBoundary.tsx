import React, { ReactElement } from 'react';
import './style.scss';

type PropsType = { children: ReactElement };
type StateType = { hasError: boolean };

class ErrorBoundary extends React.Component<PropsType, StateType> {
  constructor(props: PropsType) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError(): { hasError: boolean } {
    return { hasError: true };
  }

  render(): ReactElement {
    const { hasError } = this.state;
    const { children } = this.props;

    if (hasError) {
      return (
        <div className="error-message-block content">
          <h1>Something went wrong :(</h1>
        </div>
      );
    }

    return children;
  }
}

export default ErrorBoundary;
