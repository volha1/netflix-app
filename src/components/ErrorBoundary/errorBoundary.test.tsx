import React from 'react';
import renderer from 'react-test-renderer';

import ErrorBoundary from './ErrorBoundary';

describe('ErrorBoundary test', () => {
  it('ErrorBoundary snapshot', () => {
    const component = renderer.create(
      <ErrorBoundary>
        <div>test</div>
      </ErrorBoundary>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
