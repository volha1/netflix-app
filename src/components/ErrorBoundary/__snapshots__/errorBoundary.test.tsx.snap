// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary test ErrorBoundary snapshot 1`] = `
<div>
  test
</div>
`;
